Dockerfile

```diff
@@ -0,0 +1,42 @@
+###############################################
+# Base Image
+###############################################
+FROM python:3.8.10 as python-base
+
+ENV PYTHONUNBUFFERED=1 
+ENV PYTHONDONTWRITEBYTECODE=1 
+ENV PIP_NO_CACHE_DIR=off 
+ENV PIP_DISABLE_PIP_VERSION_CHECK=on 
+ENV PIP_DEFAULT_TIMEOUT=100 
+ENV POETRY_VERSION=1.1.12 
+ENV POETRY_HOME="/opt/poetry" 
+ENV POETRY_VIRTUALENVS_IN_PROJECT=true 
+ENV POETRY_NO_INTERACTION=1 
+ENV PYSETUP_PATH="/opt/pysetup" 
+ENV VENV_PATH="/opt/pysetup/.venv"
+
+# prepend poetry and venv to path
+ENV PATH="$POETRY_HOME/bin:$VENV_PATH/bin:$PATH"
```

Comment:
@lildonpancho i cobbled together this Dockerfile. What do you think?

Reply:
my understanding is that the first build is just compiling the python and that is passed to the second build. So this our only win in terms of size for the multistage build and files from that first stage are not required. Could be totally misunderstanding this

Reply:
Sorry I didn't get to this sooner. I think this [link](https://pythonspeed.com/articles/multi-stage-docker-python/) might clear some things up for you.